pkg/automationconfig/automation_config.go

```diff
@@ -60,6 +70,22 @@ func (p *Process) SetPort(port int) *Process {
 	return p.SetArgs26Field("net.port", port)
 }
 
+func (p *Process) GetPort() int {
+	if p.Args26 == nil {
+		return 0
+	}
+
+	// Args26 map could be manipulated in memory, e.g. via SetPort - then it will be as int,
```

Comment:
It's the second time I'm reading this comment, so it makes me scratch my head - shouldn't we be using defensive copying when unmarshaling the configuration?

Reply:
What do you mean by "using defensive copying" here? 
The comment here is intentionally copied from MongodConfiguration. I lost too many hours chasing bugs with these ints and floats! As said before - it'd be good to get rid of ints!

Reply:
By defensive copy I mean that you replace the internal map whenever you mutate it (just clone it, update one of the values and replace it). This way, whatever caller has a reference to that map, it has a guarantee it won't change.

Reply:
:thumbsup:

Reply:
Done

Reply:
Done